core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -328,6 +328,10 @@ public void request(int numMessages) {
 
   @Override
   public void cancel(@Nullable String message, @Nullable Throwable cause) {
+    if (message == null && cause == null) {
+      cause = new CancellationException("Cancelled without a message or cause");
+      log.log(Level.WARNING, "Cancelling without a message or cause is suboptimal", cause);
```

Comment:
Oh, oops. I see now how "suboptimal" was explained by the exception. Sorry. It was fine before, but this works too.